src/main/scala/is/hail/methods/UpperIndexBounds.scala

```diff
@@ -0,0 +1,65 @@
+package is.hail.methods
+
+import is.hail.HailContext
+import is.hail.annotations.UnsafeIndexedSeq
+import is.hail.expr.types._
+import is.hail.linalg.BlockMatrix
+import is.hail.table.Table
+import is.hail.utils.fatal
+import org.apache.spark.sql.Row
+
+object UpperIndexBounds {
+
+  def groupPositionsByContig(tbl: Table): Array[Array[Int]] = {
+    val expectedSignature = TStruct("contig" -> TString(), "pos" -> TInt32())
```

Comment:
I don't think the names and key type matter. Is there any constraint other than there being exactly one value field, and that that field has type `TInt32()`?